ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/CatIndexTool.java

```diff
@@ -73,80 +70,98 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        List<String> indexList = gson.fromJson(parameters.get("indices"), List.class);
-        String[] indices = parameters.containsKey("indices")? indexList.toArray(new String[0]) : new String[]{};
-
+        String[] indices = null;
+        if (parameters.containsKey("indices")) {
+            @SuppressWarnings("unchecked")
+            List<String> indexList = gson.fromJson(parameters.get("indices"), List.class);
+            indices = indexList.toArray(new String[0]);
```

Comment:
what happens if a request comes to cat an system index?

Reply:
Huh. Looks like I need to use the getSettings approach [here](https://github.com/opensearch-project/OpenSearch/blob/8673fa937db405b8d614f8d4a02c0aa52587c037/server/src/main/java/org/opensearch/rest/action/cat/RestIndicesAction.java#L146)

Reply:
Fixed to match REST API